File: server/data/results/translations/plain/go/run_length_encode.go
package main

import (
	"fmt"
	"strconv"
)

func encode(text string) string {
	if len(text) == 0 {
		return ""
	}
	result := ""
	current := text[0]
	count := 1
	i := 1
	for i < len(text) {
		if text[i] == current {
			count = count + 1
		} else {
			result = result + string(current) + strconv.Itoa(count)
			current = text[i]
			count = 1
		}
		i = i + 1
	}
	result = result + string(current) + strconv.Itoa(count)
	return result
}

func main() {
	for _, sample := range []string{"aaabbc", "abcd", "zzzzzzzz"} {
		fmt.Println(encode(sample))
	}
}

File: server/data/results/translations/plain/cpp/run_length_encode.cpp
#include <iostream>
#include <string>
#include <vector>

std::string encode(const std::string& text) {
    if (text.empty()) {
        return "";
    }
    std::string result = "";
    char current = text[0];
    int count = 1;
    size_t i = 1;
    while (i < text.length()) {
        if (text[i] == current) {
            count = count + 1;
        } else {
            result = result + current + std::to_string(count);
            current = text[i];
            count = 1;
        }
        i = i + 1;
    }
    result = result + current + std::to_string(count);
    return result;
}

int main() {
    std::vector<std::string> samples = {"aaabbc", "abcd", "zzzzzzzz"};
    for (const auto& sample : samples) {
        std::cout << encode(sample) << std::endl;
    }
    return 0;
}